api/trace/b3_propagator.go

```diff
@@ -82,6 +82,9 @@ func (b3 B3) Extract(ctx context.Context, supplier propagation.HTTPSupplier) con
 	} else {
 		sc = b3.extract(supplier)
 	}
+	if !sc.IsValid() {
```

Comment:
If instead we had `B3.extractSingleHeader` and `B3.extract` return a bool value as a second return value, we could avoid the byte array comparison in `TraceID.IsValid`. Did you consider that alternative?

Reply:
Same here.